portfolio/src/main/java/com/google/sps/data/BlogHashMap.java

```diff
@@ -60,26 +61,26 @@ public void putInMap(List<BlogMessage> messages) {
 
   // Returns a requested amount of all BlogMessages in map
   private LinkedList<BlogMessage> getMessages(int limit) {
-    LinkedList values = new LinkedList();
-
-    for (LinkedList<BlogMessage> tagMessages : map.values()){
+    LinkedList<BlogMessage> values = new LinkedList<BlogMessage>();
+
+    // |valuesList| is a list of LinkedList BlogMessages for each tag in map, hence
+    // We reverse iterate starting from the last tag entered.
+    List valuesList = new ArrayList(map.values());
+    ListIterator reverseItr = valuesList.listIterator(valuesList.size());
```

Comment:
Similarly use ListIterator<LinkedLists<String>> reverseItr so that you don't need the type conversion in line 76.